lib/Sema/ResilienceDiagnostics.cpp

```diff
@@ -78,6 +78,20 @@ bool TypeChecker::diagnoseInlinableDeclRefAccess(SourceLoc loc,
   // Remember that the module defining the decl must be imported publicly.
   recordRequiredImportAccessLevelForDecl(D, DC, AccessLevel::Public, loc);
 
+  // Accessing a private setter from inlinable function is deprecated
+  if (auto *accessor = dyn_cast<AccessorDecl>(D)) {
+    if (accessor->getAccessorKind() == AccessorKind::Set &&
+        accessor->getFormalAccess() == AccessLevel::Private) {
+      auto *storage = accessor->getStorage();
+      if (storage->getAttrs().hasAttribute<UsableFromInlineAttr>()) {
```

Comment:
What if the storage is public? We don't want to check for this attribute directly, instead check it using access scopes like we do for everything else.